### PR TITLE
added df oneAPI tests (catch2) for classification/regression train/infer

### PR DESCRIPTION
**Pull request by makart19**

### CODE REVIEW

File: cpp/oneapi/dal/algo/decision_forest/BUILD [hunk 1]
```diff
@@ -74,3 +74,17 @@ dal_test_suite(
         ":gpu_tests",
     ],
 )
+
+dal_test_suite(
+    name = "interface_cls_tests",
```
Comment (amgrigoriev) [inline cpp/oneapi/dal/algo/decision_forest/BUILD:79]: interface_tests?

File: cpp/oneapi/dal/algo/decision_forest/BUILD [hunk 2]
```diff
@@ -74,3 +74,17 @@ dal_test_suite(
...
+    framework = "catch2",
+    srcs = glob([
+        "test/batch_classification.cpp",
+    ]),
+    dal_deps = [
+        ":decision_forest",
+    ],
+    tags = [
```
Comment (amgrigoriev) [inline cpp/oneapi/dal/algo/decision_forest/BUILD:87]: Remove?

File: cpp/oneapi/dal/algo/decision_forest/BUILD [hunk 3]
```diff
@@ -74,3 +74,17 @@ dal_test_suite(
...
+        "interface",
+    ]
+)
```
Comment (amgrigoriev) [inline cpp/oneapi/dal/algo/decision_forest/BUILD:90]: Add? 
dal_test_suite(
    name = "tests",
    tests = [
        ":interface_tests",
    ],
)

File: cpp/oneapi/dal/algo/decision_forest/test/batch_classification.cpp [hunk 4]
```diff
@@ -0,0 +1,527 @@
...
+                    default: row_ind = 0; break;
+                };
+
+                const auto var_imp_val =
+                    dal::row_accessor<const Float>(train_result.get_var_importance()).pull();
+                const auto required_var_imp_val =
+                    dal::row_accessor<const float>(required_var_imp).pull({ row_ind, row_ind + 1 });
+
+                for (std::int32_t i = 0; i < var_imp_val.get_count(); i++) {
+                    REQUIRE(((required_var_imp_val[i] - var_imp_val[i]) / required_var_imp_val[i]) <
```
Comment (amgrigoriev) [inline cpp/oneapi/dal/algo/decision_forest/test/batch_classification.cpp:157]: Is required_var_imp_val[i] always not equal to zero?

File: cpp/oneapi/dal/algo/decision_forest/test/batch_classification.cpp [hunk 5]
```diff
@@ -0,0 +1,527 @@
...
+
+    void check_oob_err_matches_required(const df::descriptor<Float, Method, Task>& desc,
+                                        const df::train_result<Task>& train_result,
+                                        double required_oob_error,
+                                        double accuracy_threshold) {
+        if (check_mask_flag(desc.get_error_metric_mode(), error_metric_mode::out_of_bag_error)) {
+            SECTION("match of oob error vs required one is expected") {
+                const auto oob_err_val =
+                    dal::row_accessor<const double>(train_result.get_oob_err()).pull();
+                REQUIRE(std::abs((required_oob_error - oob_err_val[0]) / required_oob_error) <
```
Comment (amgrigoriev) [inline cpp/oneapi/dal/algo/decision_forest/test/batch_classification.cpp:172]: Is required_oob_error != 0?

File: cpp/oneapi/dal/algo/decision_forest/test/batch_classification.cpp [hunk 6]
```diff
@@ -0,0 +1,527 @@
...
+                    if (oob_err_per_obs_arr[i] >= 0.0) {
+                        oob_err_obs_count++;
+                        ref_oob_err += oob_err_per_obs_arr[i];
+                    }
+                    else {
+                        REQUIRE(oob_err_per_obs_arr[i] >= -1.0);
+                    }
+                }
+
+                REQUIRE(((oob_err_val[0] - ref_oob_err) / oob_err_val[0]) <
```
Comment (amgrigoriev) [inline cpp/oneapi/dal/algo/decision_forest/test/batch_classification.cpp:203]: oob_err_val[0] != 0?

File: cpp/oneapi/dal/algo/decision_forest/BUILD (hunk 1 above)
Comment (makart19) [reply cpp/oneapi/dal/algo/decision_forest/BUILD]: fixed

File: cpp/oneapi/dal/algo/decision_forest/BUILD (hunk 2 above)
Comment (makart19) [reply cpp/oneapi/dal/algo/decision_forest/BUILD]: done

File: cpp/oneapi/dal/algo/decision_forest/BUILD (hunk 3 above)
Comment (makart19) [reply cpp/oneapi/dal/algo/decision_forest/BUILD]: added

File: cpp/oneapi/dal/algo/decision_forest/test/batch_classification.cpp (hunk 4 above)
Comment (makart19) [reply cpp/oneapi/dal/algo/decision_forest/test/batch_classification.cpp]: check added

File: cpp/oneapi/dal/algo/decision_forest/test/batch_classification.cpp (hunk 5 above)
Comment (makart19) [reply cpp/oneapi/dal/algo/decision_forest/test/batch_classification.cpp]: check added

File: cpp/oneapi/dal/algo/decision_forest/test/batch_classification.cpp (hunk 6 above)
Comment (makart19) [reply cpp/oneapi/dal/algo/decision_forest/test/batch_classification.cpp]: check added

File: cpp/oneapi/dal/algo/decision_forest/test/badarg.cpp [hunk 7]
```diff
@@ -0,0 +1,295 @@
...
+    TEMPLATE_LIST_TEST_M(df_badarg_test, name, "[df][badarg]", df_cls_types)
+
+#define DF_BADARG_TEST(name) \
+    TEMPLATE_LIST_TEST_M(df_badarg_test, name, "[df][badarg]", df_common_types)
+
+DF_BADARG_CLS_TEST("throws if class_count is le 0") {
+    SKIP_IF(this->not_available_on_device());
+    REQUIRE_THROWS_AS(this->get_default_descriptor().set_class_count(0), domain_error);
+    REQUIRE_THROWS_AS(this->get_default_descriptor().set_class_count(-1), domain_error);
+}
```
[the remark this answers is not in the document]
Comment (makart19) [reply cpp/oneapi/dal/algo/decision_forest/test/badarg.cpp]: fixed

File: cpp/oneapi/dal/algo/decision_forest/backend/gpu/infer_train_cls_kernel_dpc_test.cpp [hunk 8]
```diff
@@ -15,7 +15,6 @@
 *******************************************************************************/
 
 #include <CL/sycl.hpp>
-#include <iostream>
 
 #include "gtest/gtest.h"
```
[the remark this answers is not in the document]
Comment (makart19) [reply cpp/oneapi/dal/algo/decision_forest/backend/gpu/infer_train_cls_kernel_dpc_test.cpp]: file was removed

File: cpp/oneapi/dal/algo/decision_forest/backend/gpu/infer_train_cls_kernel_dpc_test.cpp [hunk 9]
```diff
@@ -91,7 +90,7 @@ TEST(df_bad_arg_tests, test_overflow_checks_in_train) {
                              df_hist_classifier{}.set_tree_count(0x7FFFFFFFFFFFFFFF),
                              x_train_table,
                              y_train_table)),
-                 dal::domain_error);
+                 dal::internal_error);
```
[the remark this answers is not in the document]
Comment (makart19) [reply cpp/oneapi/dal/algo/decision_forest/backend/gpu/infer_train_cls_kernel_dpc_test.cpp]: file was removed

**Review by amgrigoriev — APPROVED** [review]

**Pull request merged**